src/baldr/edgeinfo.cc

```diff
@@ -32,6 +32,36 @@ json::ArrayPtr names_json(const std::vector<std::string>& names) {
   return a;
 }
 
+// per tag parser. each returned string includes the leading TaggedValue.
+std::vector<std::string> parse_tagged_value(const char* ptr) {
+  switch (static_cast<TaggedValue>(ptr[0])) {
+    case TaggedValue::kLayer:
+    case TaggedValue::kBssInfo:
+    case TaggedValue::kLevel:
+    case TaggedValue::kLevelRef:
+    case TaggedValue::kTunnel:
+    case TaggedValue::kBridge:
+      return {std::string(ptr)};
+    case TaggedValue::kPronunciation: {
+      std::vector<std::string> names{};
+      size_t pos = 1;
+      while (pos < strlen(ptr)) {
+        const auto header = valhalla::midgard::unaligned_read<linguistic_text_header_t>(ptr + pos);
+        pos += 3;
+        names.emplace_back((std::string(reinterpret_cast<const char*>(&header), 3) +
+                            std::string((ptr + pos), header.length_)));
+
+        pos += header.length_;
+      }
+      return names;
+    }
+    case TaggedValue::kLandmark:
+      return {std::string(ptr, find_landmark_taggedvalue_end(ptr))};
```

Comment:
just bring the function from the header over here

Reply:
oops sorry I forgot to do that. not good at remembering things...